lib/rules/echidna/deliverer-change.js

```diff
@@ -0,0 +1,59 @@
+import w3cApi from 'node-w3capi';
+
+w3cApi.apiKey = process.env.W3C_API_KEY;
+
+const self = {
+    name: 'echidna.deliverer-change',
+    section: 'metadata',
+    rule: 'delivererID',
+};
+
+export const { name } = self;
+
+function getPreviousDelivererIDs(shortname, previousUrl) {
+    const date = previousUrl.match(/-(\d{8})\/$/);
+    if (date)
+        return new Promise(resolve => {
+            // e.g. https://api.w3.org/specifications/WGSL/versions/20230221/deliverers
+            w3cApi
+                .specification(shortname)
+                .version(date[1])
+                .deliverers()
+                .fetch({ embed: true }, (err, data) => {
+                    resolve(data && data.map(obj => obj.id));
+                });
+        });
+    return [];
```

Comment:
This is wrong. In one case the function returns a Promise and in the other, it returns an empty array...
That function should return a promise for each case...